fe/fe-core/src/main/java/com/starrocks/catalog/Column.java

```diff
@@ -757,6 +757,40 @@ public boolean equals(Object obj) {
         return comment == null ? other.comment == null : comment.equals(other.getComment());
     }
 
+    public boolean isSchemaCompatible(Column other) {
+        if (!this.name.equalsIgnoreCase(other.getName())) {
+            return false;
+        }
+        if (!this.getType().equals(other.getType())) {
+            return false;
+        }
+        if (!(aggregationType == other.aggregationType || (AggregateType.isNullOrNone(aggregationType) &&
+                AggregateType.isNullOrNone(other.getAggregationType())))) {
+            return false;
+        }
+        if (this.isAggregationTypeImplicit != other.isAggregationTypeImplicit()) {
+            return false;
+        }
+        if (this.isKey != other.isKey()) {
+            return false;
+        }
+        if (this.isAllowNull != other.isAllowNull) {
+            return false;
+        }
+        if (this.getType().isScalarType() && other.getType().isScalarType()) {
```

Comment:
If one is struct<f1 char(10)> and the other is struct<f2 char(20)>, this function will return true.
but if one is f1 char(10) and the other is f2 char(20), this function will return false.
it is strange to users.

Reply:
both of them will return false:
1. struct type: recursive check all fields and `type.equals(other.type) => false`
2. scalar type: `type.equals(other.type) => false`